state/accountsRepository.go

```diff
@@ -2,62 +2,109 @@ package state
 
 import (
 	"github.com/ElrondNetwork/elrond-go-core/core/check"
+	chainData "github.com/ElrondNetwork/elrond-go-core/data"
 	vmcommon "github.com/ElrondNetwork/elrond-vm-common"
 )
 
-// >> BEGIN notes for review
-// "accountsDBApi" could not be used because it only allows its creator (caller of constructor) to control the provisioning of the "targetRootHash".
-// Being constrained by the AccountsAdapter interface, it does not allow the caller of its methods to specify the "targetRootHash".
-// > Design constraint: we have to return the nonce & hash associated with the rootHash (on GET account?onFinalBlock=true), as well.
-// > Design constraint: the one that provides the "targetRootHash" must also access in a consistent manner (e.g. critical section) the block nonce & block hash associated with the rootHash,
-// so that they are paired (in a consistent manner) with the loaded account data.
-//
-// Having "accountsDBApi" to use blockchain.GetFinalBlockInfo().rootHash to load the account data, then having the "nodeFacade" to return the block nonce and block hash (also from the chainHandler)
-// would have possibly resulted in occasional inconsistencies. Instead, we'll make the "nodeFacade" responsible to call blockchain.GetFinalBlockInfo(), hold the results,
-// call accountsRepository with the returned rootHash etc.
-// << END notes for review
-
-// Question for review: perhaps rename to "accountsByRootHashRepository"?
 type accountsRepository struct {
-	innerAccountsAdapter AccountsAdapter
-	trieController       *accountsDBApiTrieController
+	chainHandler       chainData.ChainHandler
+	strategyForFinal   *accountsRepositoryStrategy
```

Comment:
rename to `finalAccountsRepository` and `currentAccountsRepository` ?

Reply:
Not applicable anymore.